lib/lib.misc.php

```diff
@@ -1386,28 +1445,35 @@ function eventlog($type, $dataid, $action, $cid = null, $json = null, $id = null
 	$now = sprintf('%.3f', microtime(TRUE));
 
 	// TODO: can this be wrapped into a single query?
-	$eventids = array();
+	$eventids = [];
 	foreach ( $cids as $cid ) {
 		$table = ( $endpoint['tables'] ? $endpoint['tables'][0] : NULL );
-		$eventid = $DB->q('RETURNID INSERT INTO event
-		                   (eventtime, cid, endpointtype, endpointid,
-		                   datatype, dataid, action, content)
-		                   VALUES (%s, %i, %s, %s, %s, %s, %s, %s)',
-		                  $now, $cid, $type, $id,
-		                  $table, $dataid, $action, $json);
-		$eventids[] = $eventid;
+		foreach ($dataids as $idx => $dataid) {
+			if ($multiple) {
+				$jsonElement = dj_json_encode($json[$idx]);
+			} else {
+				$jsonElement = dj_json_encode($json);
+			}
+			$eventid = $DB->q('RETURNID INSERT INTO event
```

Comment:
Can we do one bigger DB query here instead?

Reply:
There is already a TODO above here.
It might be possible (although I don't know how we are getting the RETURNID then), but I think this is not for 6.0 for now.